dot/rpc/http_test.go

```diff
@@ -90,3 +90,45 @@ func TestNewHTTPServer(t *testing.T) {
 
 	require.Equal(t, "405 Method Not Allowed", res.Status)
 }
+
+func TestBlockExternalIP(t *testing.T) {
+	coreAPI := core.NewTestService(t, nil, nil)
+	si := &types.SystemInfo{
+		SystemName: "gossamer",
+	}
+	sysAPI := system.NewService(si)
+	cfg := &HTTPServerConfig{
+		Modules:    []string{"system"},
+		RPCPort:    8545,
+		RPCAPI:     NewService(),
+		CoreAPI:    coreAPI,
+		SystemAPI:  sysAPI,
+		External:   false,
+		WSExternal: false,
+	}
+
+	s := NewHTTPServer(cfg)
+	err := s.Start()
+	require.Nil(t, err)
+
+	time.Sleep(time.Second) // give server a second to start
+
+	// Valid request
+	client := &http.Client{}
+	data := []byte(`{"jsonrpc":"2.0","method":"system_name","params":[],"id":1}`)
+
+	buf := &bytes.Buffer{}
+	_, err = buf.Write(data)
+	require.Nil(t, err)
+	req, err := http.NewRequest("POST", "http://localhost:8545/", buf)
+	require.Nil(t, err)
+
+	req.Header.Set("Content-Type", "application/json")
+
+	res, err := client.Do(req)
+	require.Nil(t, err)
+	defer res.Body.Close()
+
+	require.Equal(t, "200 OK", res.Status)
+
```

Comment:
This test is called TestBlockExternalIP, but it doesn't seem to be testing an external IP. It does seem to confirm a local connection works. I see `server.rpcServer.RegisterValidateRequestFunc(LocalRequestOnly)` which seem like it's doing what it should, but I don't see how this is tested.

Reply:
Ahh good spot, this was my initial set up test but wasn't sure how to simulate external IP addresses, removing for now